### PR TITLE
Remove pin of opentelemetry-api from azure-core-tracing-opentelemetry

### PR DESCRIPTION
**Pull request by lzchen**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/31965

Due to the now released version of: https://pypi.org/project/opentelemetry-instrumentation-requests/, the below combination of libraries should be able to be successfully installed.

```
opentelemetry-api==v1.20.0
opentelemetry-api==v1.20.0
opentelemetry-instrumentation-requests==v0.41b0
```

### CODE REVIEW

**Review by scbedd — APPROVED** [review]
Seems reasonable! I'm ok with approving and seeing if `latestdependency` still checks out!

Thanks for the heads up @lzchen !

**Comment by scbedd** [comment]
I just kicked the [internal build](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3238073&view=results) using this code. If that build is green, it's definitely g2g.

The PR checks don't include `latestdependency` by default. it's `sdist, whl, depends, mindependency`

**Comment by scbedd** [comment]
Some of the `core` latestdependency envs passed, you're good to merge this @lzchen . thanks for the PR!

**Comment by lzchen** [comment]
@scbedd 

Any idea what the failing tests are for mypy for azure-core? Looks like it's blocking merging.

**Comment by lzchen** [comment]
Overriding. All is green except for a mypy related error in azure-core that is irrelevant to current change.

**Pull request merged**
